mautic/mautic
# Fix migration for 3.3.0

| Q                                      | A
| -------------------------------------- | ---
| Branch?                                | "3.3" 
| Bug fix?                               | no
| New feature?                           |no
| Deprecations?                          |no
| BC breaks?                             |no
| Automated tests included?              |no
| Related user documentation PR URL      | na
| Related developer documentation PR URL | na
| Issue(s) addressed                     | Fixes #9709

#### Description:
This addresses schema differences for M2 and M3 introduced in https://github.com/mautic/mautic/pull/9544. It should also remove the need for https://github.com/mautic/mautic/pull/9714 since it seems that some categories.id has a length of 10 and some 11 for different versions of MariaDB. 

#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com)
2. Preferably run the migrations on a failed upgrade, a M2  installation upgraded to M3, and an installation on M3. All three should work. (I've only tested on a M3 schema as I don't have a M2 schema but tried to manipulate the database to emulate all three scenarios).
